internal/operands/common-templates/reconcile_test.go

```diff
@@ -69,35 +75,67 @@ var _ = Describe("Common-Templates operand", func() {
+	})
+
+	It("should create golden-images namespace", func() {
 		ExpectResourceExists(newGoldenImagesNS(GoldenImagesNSname), request)
 	})
 	It("should create common-template resources", func() {
-		_, err := operand.Reconcile(&request)
-		Expect(err).ToNot(HaveOccurred())
 		Expect(templatesBundle).ToNot(BeNil())
 		for _, template := range templatesBundle {
 			template.Namespace = namespace
 			ExpectResourceExists(&template, request)
 		}
 	})
 	It("should create view role", func() {
-		_, err := operand.Reconcile(&request)
-		Expect(err).ToNot(HaveOccurred())
 		ExpectResourceExists(newViewRole(GoldenImagesNSname), request)
 	})
 	It("should create view role binding", func() {
-		_, err := operand.Reconcile(&request)
-		Expect(err).ToNot(HaveOccurred())
 		ExpectResourceExists(newViewRoleBinding(GoldenImagesNSname), request)
 	})
 	It("should create view role binding", func() {
-		_, err := operand.Reconcile(&request)
-		Expect(err).ToNot(HaveOccurred())
 		ExpectResourceExists(newEditRole(), request)
 	})
+
+	It("should remove labels from old templates", func() {
+		var oldTemplates templatev1.TemplateList
+		versionRequirement, err := labels.NewRequirement("template.kubevirt.io/version", selection.NotEquals, []string{Version})
+		Expect(err).ToNot(HaveOccurred())
+		labelsSelector := labels.NewSelector().Add(*versionRequirement)
+		opts := client.ListOptions{
+			LabelSelector: labelsSelector,
+		}
+		Expect(request.Client.List(request.Context, &oldTemplates, &opts)).ToNot(HaveOccurred())
+		for _, template := range oldTemplates.Items {
+			Expect(template.Labels["os.template.kubevirt.io/some-os"]).To(Equal(""), "os.template.kubevirt.io should be empty")
+			Expect(template.Labels["flavor.template.kubevirt.io/test"]).To(Equal(""), "flavor.template.kubevirt.io should be empty")
+			Expect(template.Labels["workload.template.kubevirt.io/server"]).To(Equal(""), "workload.template.kubevirt.io should be empty")
+			Expect(template.Labels["template.kubevirt.io/type"]).To(Equal("base"), "template.kubevirt.io/type should equal base")
+			Expect(template.Labels["template.kubevirt.io/version"]).To(Equal(oldTemplateVersion), "template.kubevirt.io/version should equal "+oldTemplateVersion)
+		}
+	})
+	It("should not remove labels from latest templates", func() {
+		var oldTemplates templatev1.TemplateList
+		versionRequirement, err := labels.NewRequirement("template.kubevirt.io/version", selection.Equals, []string{Version})
+		Expect(err).ToNot(HaveOccurred())
+		labelsSelector := labels.NewSelector().Add(*versionRequirement)
+		opts := client.ListOptions{
+			LabelSelector: labelsSelector,
+		}
+		Expect(request.Client.List(request.Context, &oldTemplates, &opts)).ToNot(HaveOccurred())
+		for _, template := range oldTemplates.Items {
+			Expect(template.Labels["os.template.kubevirt.io/some-os"]).To(Equal("true"), "os.template.kubevirt.io should not be empty")
```

Comment:
Does this test pass? I don't see where you create a new template with those labels

Reply:
It was created here https://github.com/kubevirt/ssp-operator/pull/71/files#diff-6c919a0ae6f12f061dafb6d58a1d1c60cf6717bdfc2ab7c924ebf6a18a439939R39
But was redundant because those labels do not apply to latest and actual templates
Updated now